step-01/js/main.js

```diff
@@ -1,26 +1,73 @@
 'use strict';
 
-navigator.getUserMedia = navigator.getUserMedia ||
-    navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+// Define navigator.mediaDevices.getUserMedia() for older devices.
 
-var constraints = {
+// Old browsers might not implement mediaDevices, so we set an empty object.
+if (navigator.mediaDevices === undefined) navigator.mediaDevices = {};
+
+// Add mediaDevices.getUserMedia method if it's missing.
+if (navigator.mediaDevices.getUserMedia === undefined) {
+
+  /**
+   * Gets a stream of media content containing tracks of video or audio.
+   * Produces a Promise which resolves into a MediaStream object.
+   * @param {!MediaStreamConstraints} constraints to apply to the media stream.
+   * @return {!Promise} Promise which resolves to a MediaStream.
+   */
+  navigator.mediaDevices.getUserMedia = (constraints) => {
+    // Leverage legacy getUserMedia, if it is defined.
+    const getUserMedia = navigator.getUserMedia ||
```

Comment:
I don't think we need all that now: see demo at [simpl.info/gum](https://simpl.info/getusermedia), which works in all target browsers.

Reply:
Removed retro-compatibility code.